app/Models/BeatmapsetEvent.php

```diff
@@ -106,6 +106,13 @@ public static function search($rawParams = [])
             }
         }
 
+        if (present($rawParams['beatmapset_id'] ?? null)) {
+            $params['beatmapset_id'] = $rawParams['beatmapset_id'];
+            $beatmapset = Beatmapset::find($params['beatmapset_id']);
+
+            $query->where('beatmapset_id', '=', $beatmapset->getKey());
```

Comment:
```suggestion
            $query->where('beatmapset_id', '=', $params['beatmapset_id']);
```
otherwise it'll error when the beatmapset id doesn't exist. Also this way skips additional query.